### PR TITLE
ign -> gz Environment Variable Migration : gz-transport

### PR DESCRIPTION
**Pull request by methylDragon**
See: https://github.com/gazebo-tooling/release-tools/issues/734

### CODE REVIEW

File: src/Discovery_TEST.cc [hunk 1]
```diff
@@ -528,23 +528,23 @@ TEST(DiscoveryTest, TestActivity)
 }
 
 //////////////////////////////////////////////////
-/// \brief Check that a wrong IGN_IP value makes HostAddr() to return 127.0.0.1
-TEST(DiscoveryTest, WrongIgnIp)
+/// \brief Check that a wrong GZ_IP value makes HostAddr() to return 127.0.0.1
+TEST(DiscoveryTest, WronggzIp)
```
Comment (chapulina) [inline src/Discovery_TEST.cc:532]:
```suggestion
TEST(DiscoveryTest, WrongGzIp)
```

File: src/NodeOptions_TEST.cc [hunk 2]
```diff
@@ -25,17 +25,17 @@
...
+  // Set GZ_PARTITION
   std::string aPartition = "customPartition";
-  setenv("IGN_PARTITION", aPartition.c_str(), 1);
+  setenv("GZ_PARTITION", aPartition.c_str(), 1);
 
   transport::NodeOptions opts;
   EXPECT_EQ(opts.Partition(), aPartition);
 
-  // A partition set by the user should overwrite IGN_PARTITION.
+  // A partition set by the user should overwrite .
```
Comment (chapulina) [inline src/NodeOptions_TEST.cc:38]:
```suggestion
  // A partition set by the user should overwrite GZ_PARTITION.
```

File: src/NodeShared.cc [hunk 3]
```diff
@@ -216,25 +232,59 @@ NodeShared::NodeShared()
...
   // Set the port used for msg discovery.
   this->msgDiscPort = this->dataPtr->NonNegativeEnvVar(
-    "IGN_DISCOVERY_MSG_PORT", this->kDefaultMsgDiscPort);
+    "GZ_DISCOVERY_MSG_PORT", this->kDefaultMsgDiscPort);
+
+  // TODO(CH3): Deprecated. Remove on tock.
+  if (this->msgDiscPort == this->kDefaultMsgDiscPort)
+  {
+    this->msgDiscPort = this->dataPtr->NonNegativeEnvVar(
+      "IGN_DISCOVERY_MSG_PORT", this->kDefaultMsgDiscPort);
```
Comment (chapulina) [inline src/NodeShared.cc:274]: Is a warning printed when this is used? Same for other vars below.

Comment (methylDragon) [reply src/NodeShared.cc:274]: Nope, let me add in a conditional

Comment (methylDragon) [reply src/NodeShared.cc:274]: https://github.com/gazebosim/gz-transport/pull/318/commits/aeb86bd1c4411d725681a8ffaf7b93ffb511b815

I also moved the couts to cerr in this commit

**Review by chapulina — APPROVED** [review]
Nice! LGTM

**Pull request merged**